tutoraspects/templates/aspects/apps/superset/pythonpath/queries/dim_courses.sql

```diff
@@ -0,0 +1,23 @@
+{%- raw -%}
+
+select distinct
+    org,
+    course_key,
+    display_name as course_name,
+    location as run_id,
+    JSON_VALUE(xblock_data_json, '$.run') as run_name
+from
+    {{ dataset(18) }}
+where
+    JSON_VALUE(xblock_data_json, '$.block_type') = 'course'
+    {% if filter_values('org') != [] %}
+    and org in {{ filter_values('org') | where_in }}
+    {% endif %}
+    {% if filter_values('course_name') != [] %}
+    and display_name in {{ filter_values('course_name') | where_in }}
+    {% endif %}
+    {% if filter_values('run') != [] %}
+    and run_name in {{ filter_values('run') | where_in }}
+    {% endif %}
+
+{%- endraw -%}
```

Comment:
Is this not added automatically by the filter once you link it?

Reply:
The filters are added to the query as a whole, but ClickHouse doesn't use indexes for the right-hand side table in joins. This dataset — and the additional filters added here — are a way to limit the data that's selected before trying to join it to the larger event tables. If and when ClickHouse adds the ability to use indexes on both tables in a join, this will be unnecessary, but at the moment I think queries will fail if we try to join display names into the dataset.

Reply:
Thanks